### PR TITLE
added patch routes endpoint

### LINKED ISSUE #162: Make Routes Mutable

### PR DESCRIPTION
**Pull request by NorthFury**
Closes #162

### CODE REVIEW

File: src/main/scala/org/zalando/spearheads/innkeeper/routes/PatchRoutes.scala [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      val reqDesc = "patch /routes"
+      logger.info(s"try to $reqDesc")
+
+      entity(as[RoutePatch]) { routePatch =>
+        logger.info(s"We try to $reqDesc unmarshalled routePatch $routePatch")
+
+        team(authenticatedUser, token, reqDesc) { team =>
+          logger.debug(s"patch /routes team $team")
+
+          findPathByRouteId(id, pathsService, "delete /routes/{}")(executionContext) { path =>
```
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/routes/PatchRoutes.scala]: hopefully no delete happens here :) (and on the next line)

File: src/it/scala/org/zalando/spearheads/innkeeper/routes/PatchRoutesSpec.scala [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import org.scalatest.{BeforeAndAfter, FunSpec, Matchers}
+import org.zalando.spearheads.innkeeper.api.JsonProtocols._
+import org.zalando.spearheads.innkeeper.api._
+import org.zalando.spearheads.innkeeper.routes.AcceptanceSpecToken._
+import org.zalando.spearheads.innkeeper.routes.AcceptanceSpecsHelper._
+import spray.json._
+
+class PatchRoutesSpec extends FunSpec with BeforeAndAfter with Matchers {
+
+  private val newDescription = "new-description"
```
Comment (danpersa) [inline src/it/scala/org/zalando/spearheads/innkeeper/routes/PatchRoutesSpec.scala]: I would put the private vals at the end of the class (as we already have them in other tests)
The motivation would be that the important stuff are the tests, and not the mock data.

**Comment by NorthFury** [comment]
fixed

**Comment by danpersa** [comment]
👍

**Pull request merged**